core/block_processor.go

```diff
@@ -81,8 +81,10 @@ func (sm *BlockProcessor) TransitionState(statedb *state.StateDB, parent, block
 	return receipts, nil
 }
 
-func (self *BlockProcessor) ApplyTransaction(coinbase *state.StateObject, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
-	cb := statedb.GetStateObject(coinbase.Address())
+func (self *BlockProcessor) ApplyTransaction(statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *big.Int, transientProcess bool) (*types.Receipt, *big.Int, error) {
+	cb := statedb.GetOrNewStateObject(header.Coinbase)
+	cb.SetGasLimit(header.GasLimit)
```

Comment:
It's done at beginning of block processing before first tx is processed: https://github.com/ethereum/go-ethereum/blob/fd512fa12c59657d9e47cc3411e6e24bd1af89cb/core/block_processor.go#L73

It's used in further checks - though now that I think of it, we have to ensure that we do not reset how much gas has been used so far in the block, as the block's gasLimit applies for the full set of txs.